MyJavaProject/src/leetcodeproblems/CalculatePower.java

```diff
@@ -0,0 +1,44 @@
+import java.lang.Double;
+/* 
+Overflowing value : when n = 2^31 = 2147483648
+*/
+class CalculatePower {
+    public double myPow(double x, int n) {
+        boolean overflow_risk = false;
+        boolean neg_power = false;
+        double product, temp = 0.0;
+        if(x == 0){
+            return Double.POSITIVE_INFINITY;
+        }
+        // 2^31
+        if(n == -2147483648){
+            overflow_risk = true; 
+        }
+        if(n < 0){
+            neg_power = true;
+            n = n * (-1);
+        }
+        System.out.println(n);
+        if(overflow_risk && neg_power){
+            n--;
+        }
+        if(n == 0) {
+            return 1;
+        } 
+        else if (n % 2 == 0) {
+            temp = myPow(x, n/2);
```

Comment:
I can use Dynamc programming to speed up the calculation.

Keep an array of size n+1 and store values of pow[i]  where pow[i] = pow(x,i). Calculate this value from botton up.